Correct datetimes in variable_assignment_testdata.nc

wsmigaj opened the pull request:
## Description

In some strings in this file, dates were separated from times with `@`'s instead of `T`'s.
### Issue(s) addressed

Needed by https://github.com/JCSDA-internal/ufo/pull/1280.

## Dependencies

None

## Impact

None

## Test Data

None

wsmigaj committed:
Correct datetimes in variable_assignment_testdata.nc.

In some strings dates were separated from times with @'s instead of T's.

wsmigaj committed:
Merge branch 'develop' into feature/templated_obs_function

wsmigaj committed:
Merge branch 'develop' into feature/templated_obs_function

shlyaeva commented:
@rhoneyager and if we merge this one too? (This would merge https://github.com/JCSDA-internal/ufo-data/pull/25 and https://github.com/JCSDA-internal/ufo-data/pull/18 into develop).

wsmigaj committed:
Allow DrawValueFromFile to produce values of types other than float (#18)

* Made variable_assignment_testdata.nc readable by ncdump.

* Added files needed by tests of the DrawValueFromFile ObsFunction.

* Added extra test files for the DataExtractor.

* Extended variable_assignment_testdata.nc with a new variable, surface_pressure (#25)

It has some missing values both in the ObsValue and the ObsError group (at non-overlapping locations).
